src/DynamoCore/Extensions/IExtension.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Dynamo.Extensions
+{
+    /// <summary>
+    /// An extension to the model layer of Dynamo
+    /// </summary>
+    public interface IExtension : IDisposable
+    {
+        /// <summary>
+        /// A unique id for this extension instance.  
+        /// 
+        /// There may be multiple instances of the same type, but the application 
+        /// will *not* allow two instances to coexist with the same id.
+        /// </summary>
+        Guid Id { get; }
```

Comment:
Consider UniqueId.

I'm marginal about using the value of using a type here rather than a string. It is more explicit about  etc, but is annoying to store, etc.